### PR TITLE
Augment search API with reranking

### PR DESCRIPTION
**Pull request by sarafalamaki**
Added demonstration on how to extend the search API by adding reranking.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,3 +11,18 @@ The Redactive Application, Docs &amp; Samples
 `├──`[`sdks`](sdks/python/) — Redactive SDKs<br>
 `├──`[`.pre-commit-config.yaml`](.pre-commit-config.yaml) — Pre-commit hooks<br>
 `└──`[`.prettierrc.yaml`](.prettierrc.yaml) — Prettier configuration<br>
+
```
Comment (russellwmy) [inline README.md]: Can we move inside `sdks/python`

**Review by russellwmy — CHANGES_REQUESTED** [review]

File: sdks/python/pyproject.toml [hunk 2]
```diff
@@ -33,7 +38,7 @@ Source = "https://github.com/redactive-ai/redactive"
 path = "src/redactive/__about__.py"
 
 [tool.hatch.envs.hatch-test]
-extra-dependencies = ["pytest", "pytest-asyncio", "pytest-httpx"]
+extra-dependencies = ["pytest", "pytest-asyncio", "pytest-httpx", "rerankers[cross-encooder]"]
```
Comment (russellwmy) [inline sdks/python/pyproject.toml]:
```suggestion
extra-dependencies = ["pytest", "pytest-asyncio", "pytest-httpx"]
```

Comment (russellwmy) [reply sdks/python/pyproject.toml]: We may not need this

**Review by russellwmy — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -8,6 +8,6 @@ The Redactive Application, Docs &amp; Samples
 `├──`[`.github`](.github/) — GitHub configuration including CI/CD workflows<br>
 `├──`[`.vscode`](.vscode/) — VS Code configuration<br>
 `├──`[`protos`](protos/) — Redactive gRpc protocol<br>
-`├──`[`sdks`](sdks/python/) — Redactive SDKs<br>
+`├──`[`sdks`](sdks/python/) — Redactive Python SDKs<br>
```
Comment (russellwmy) [inline README.md]:
```suggestion
`├──`[`sdks`](sdks/) — Redactive SDKs<br>
```

**Review by russellwmy — CHANGES_REQUESTED** [review]

File: sdks/python/src/redactive/reranking/reranker.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        merged_results = []
+        for res in ranker_results:
+            original_chunk = fetched_chunks[res.doc_id]
+            original_chunk.relevance.similarity_score = res.score
+            merged_results.append(original_chunk)
+
+        return merged_results[:top_k]
+
+
+if __name__ == "__main__":
```
Comment (russellwmy) [inline sdks/python/src/redactive/reranking/reranker.py]: Can we put this in example code instead of add main here?

**Review by russellwmy — CHANGES_REQUESTED** [review]

File: sdks/python/src/redactive/reranking/reranker.py (hunk 4 above)
Comment (sarafalamaki) [reply sdks/python/src/redactive/reranking/reranker.py]: done

**Review by russellwmy — APPROVED** [review]
LGTM

**Pull request merged**